### ISSUE TITLE
Link fails on Linux when using clang to build posix

### ISSUE THREAD

**Issue by mcharleb** [PX4/Firmware#3019]
Something is wrong with the setup for clang as a native system compiler.  The compilation passes but the link fails.

**Comment by mcharleb** [comment]
It appears that the issue is that gcc uses collect2 as a wrapper to /usr/bin/ld and clang calls /usr/bin/ld directly and the link fails with the libraries as passed.

**Comment by mcharleb** [comment]
From http://stackoverflow.com/questions/21605720/how-can-i-find-and-test-the-actual-linker-a-compiler-is-using-on-an-arbitrary:
```
#!/usr/bin/env sh

echo; which ld; echo "...Not necessarily."; echo

for CC in cc gcc clang; do echo $CC:
  for link in collect2 ld; do
    echo 'main;' | $CC -v -x c - -o /dev/null -\#\#\# 2>&1 | grep -q $link &&
    echo 'main;' | $CC -v -x c - -o /dev/null -\#\#\# 2>&1 | grep    $link |
       sed -e "s|\(.*$link\).*|\1|" -e 's/ //g' -e 's|"||' && break
   done

  which="$(which which 2>/dev/null 1>&1)" || which=echo
  $which "$(for link in collect2 ld; do
    echo 'main;' | $CC -v -x c - -o /dev/null -\#\#\# 2>&1 | grep -q $link &&
    echo 'main;' | $CC -v -x c - -o /dev/null -\#\#\# 2>&1 | grep    $link |
   sed -e "s|\(.*$link\).*|\1|" -e 's/ //g' -e 's|"||' && break
  done)"
  echo
done
```

Output:

/usr/bin/ld
...Not necessarily.

cc:
/usr/lib/llvm-3.5/bin/clang""-cc1""-triple""x86_64-pc-linux-gnu""-emit-obj""-mrelax-all""-disable-free""-disable-llvm-verifier""-main-file-name""-""-mrelocation-model""static""-mdisable-fp-elim""-fmath-errno""-masm-verbose""-mconstructor-aliases""-munwind-tables""-fuse-init-array""-target-cpu""x86-64""-target-linker-version""2.25""-v""-dwarf-column-info""-resource-dir""/usr/lib/llvm-3.5/bin/../lib/clang/3.5.2""-internal-isystem""/usr/include/clang/3.5.2/include/""-internal-isystem""/usr/local/include""-internal-isystem""/usr/lib/llvm-3.5/bin/../lib/clang/3.5.2/include""-internal-externc-isystem""/usr/bin/../lib/gcc/x86_64-linux-gnu/4.9/include""-internal-externc-isystem""/usr/include/x86_64-linux-gnu""-internal-externc-isystem""/include""-internal-externc-isystem""/usr/include""-fdebug-compilation-dir""/local2/mnt/workspace/dronecode/Firmware/build
/usr/bin/ld""--eh-frame-hdr""-m""elf_x86_64""-dynamic-linker""/lib64/ld

gcc:
/usr/lib/gcc/x86_64-linux-gnu/4.9/collect2
/usr/lib/gcc/x86_64-linux-gnu/4.9/collect2

clang:
/usr/lib/llvm-3.5/bin/clang""-cc1""-triple""x86_64-pc-linux-gnu""-emit-obj""-mrelax-all""-disable-free""-disable-llvm-verifier""-main-file-name""-""-mrelocation-model""static""-mdisable-fp-elim""-fmath-errno""-masm-verbose""-mconstructor-aliases""-munwind-tables""-fuse-init-array""-target-cpu""x86-64""-target-linker-version""2.25""-v""-dwarf-column-info""-resource-dir""/usr/lib/llvm-3.5/bin/../lib/clang/3.5.2""-internal-isystem""/usr/include/clang/3.5.2/include/""-internal-isystem""/usr/local/include""-internal-isystem""/usr/lib/llvm-3.5/bin/../lib/clang/3.5.2/include""-internal-externc-isystem""/usr/bin/../lib/gcc/x86_64-linux-gnu/4.9/include""-internal-externc-isystem""/usr/include/x86_64-linux-gnu""-internal-externc-isystem""/include""-internal-externc-isystem""/usr/include""-fdebug-compilation-dir""/local2/mnt/workspace/dronecode/Firmware/build
/usr/bin/ld""--eh-frame-hdr""-m""elf_x86_64""-dynamic-linker""/lib64/ld

**Comment by mcharleb** [comment]
@LorenzMeier fixed in PR #3025

**Comment by LorenzMeier** [comment]
Thanks!
